### PR TITLE
#90 The server start error messages could be improved to enhance user experience

### PR DESCRIPTION
**Pull request by yhmo**

### CODE REVIEW

File: core/src/server/DBWrapper.cpp [hunk 1]
```diff
@@ -54,21 +54,24 @@ DBWrapper::StartService() {
     std::string db_slave_path;
     s = config.GetDBConfigSecondaryPath(db_slave_path);
     if (!s.ok()) {
-        return s;
+        std::cerr << s.ToString() << std::endl;
+        kill(0, SIGUSR1);
```
Comment (cydrain) [inline core/src/server/DBWrapper.cpp]: Here "return s" is for specific reason, only do so can let system print out "Milvus server exit..." and tell user Milvus server does not boot up successfully.
If use "kill()", system will exit quitely without any message.

File: core/src/server/Config.cpp [hunk 2]
```diff
@@ -363,19 +363,25 @@ Config::PrintAll() {
 Status
 Config::CheckServerConfigAddress(const std::string& value) {
     if (!ValidationUtil::ValidateIpAddress(value).ok()) {
-        return Status(SERVER_INVALID_ARGUMENT, "Invalid server config address: " + value);
+        std::string msg = "Invalid server IP address: " + value
+                          + ". Possible reason: server_config.address is invalid in server_config.yaml.";
```
Comment (cydrain) [inline core/src/server/Config.cpp]: Better to uniform the format: xxx_config.xxx in xxx.yaml is invalid
Please also check below messages.

File: core/conf/server_config.template [hunk 3]
```diff
@@ -2,9 +2,9 @@
 
 server_config:
   address: 0.0.0.0                  # milvus server ip address (IPv4)
-  port: 19530                       # port range: 1025 ~ 65534
+  port: 19530                       # milvus server port, must in range [1025, 6553]
```
Comment (jielinxu) [inline core/conf/server_config.template]: in range [1025, 65534]?

File: core/conf/server_config.template [hunk 4]
```diff
@@ -14,30 +14,30 @@ db_config:
...
   preload_table:                    # preload data at startup, '*' means load all tables, empty value means no preload
                                     # you can specify preload tables like this: table1,table2,table3
 
 metric_config:
-  enable_monitor: false             # enable monitoring or not
+  enable_monitor: false             # enable monitoring or not, must be a boolean
   collector: prometheus             # prometheus
   prometheus_config:
-    port: 8080                      # port prometheus uses to fetch metrics
+    port: 8080                      # port prometheus uses to fetch metrics, must in range [1025, 6553]
```
Comment (jielinxu) [inline core/conf/server_config.template]: range is [1025, 65534]

**Review by jielinxu — APPROVED** [review]

**Review by cydrain — APPROVED** [review]

**Pull request merged**